Alpha2

dwjanus opened the pull request:
fixes the rebase issues with alpha

siferbel committed:
Clicking issue on Get Ohm button and wallet button. Was only clickable on the text, removed padding on <button> tag and added padding on <a>

unbanksy committed:
Change slippage default to 0.5%

ohmie33 committed:
Implemented wallet token integration

Commit:
Change default slippage to .5 %

Commit:
Force cast autostake arg as boolean

Commit:
Open social links in new tab

Commit:
removed unused scaffold-eth components and pruned scss

Commit:
removed unused scaffold-eth components and pruned scss

Commit:
Convert bond discounts to links (#173)

Commit:
starting themes

Commit:
starting themes

Commit:
apollo installed and implemented for basic app state. still getting issues with circ and total supply from the graph

Commit:
apollo installed and implemented for basic app state. still getting issues with circ and total supply from the graph

Commit:
apollo installed and implemented for basic app state. still getting issues with circ and total supply from the graph

Commit:
177: Fix link colors (and show underline on active state) (#179)

* Fix link colors (and show underline on active state)
* Keep golden color on dark theme and enforce border color on hover

Zayen-X committed:
Add issue templates

Commit:
Fix wrong colors of bond links (#178)

Co-authored-by: ZayenX <84199924+lolchocotaco@users.noreply.github.com>

Zayen-X committed:
updated readme

Commit:
dashboard tiles use graph queries from app state

Commit:
dashboard tiles use graph queries from app state

ohmie33 committed:
Implement new menu design and only show add token when eth api is available

ohmie33 committed:
Fix uniswap swap link

drondin committed:
Add TheGrpah ID as constant

Commit:
added /lib and moved apollo client there, removed duplicate declarations in actions

Commit:
updated thegraph id

Commit:
deleted comment block from app actions, removed console logs, renamed graph id in constants

unbanksy committed:
Add FRAX amount in treasury to total treasury bal.

drondin committed:
Fix unstake

xh3b4sd committed:
add dependabot config file

xh3b4sd committed:
fix dependabot target branch

xh3b4sd committed:
Update dependabot.yml

xh3b4sd committed:
Update dependabot.yml

dependabot[bot] committed:
Bump @testing-library/jest-dom from 5.12.0 to 5.14.1

Bumps [@testing-library/jest-dom](https://github.com/testing-library/jest-dom) from 5.12.0 to 5.14.1.
- [Release notes](https://github.com/testing-library/jest-dom/releases)
- [Changelog](https://github.com/testing-library/jest-dom/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testing-library/jest-dom/compare/v5.12.0...v5.14.1)

---
updated-dependencies:
- dependency-name: "@testing-library/jest-dom"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @testing-library/dom from 6.16.0 to 8.0.0

Bumps [@testing-library/dom](https://github.com/testing-library/dom-testing-library) from 6.16.0 to 8.0.0.
- [Release notes](https://github.com/testing-library/dom-testing-library/releases)
- [Changelog](https://github.com/testing-library/dom-testing-library/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testing-library/dom-testing-library/compare/v6.16.0...v8.0.0)

---
updated-dependencies:
- dependency-name: "@testing-library/dom"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @testing-library/dom from 6.16.0 to 8.0.0

Bumps [@testing-library/dom](https://github.com/testing-library/dom-testing-library) from 6.16.0 to 8.0.0.
- [Release notes](https://github.com/testing-library/dom-testing-library/releases)
- [Changelog](https://github.com/testing-library/dom-testing-library/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testing-library/dom-testing-library/compare/v6.16.0...v8.0.0)

---
updated-dependencies:
- dependency-name: "@testing-library/dom"
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @ethersproject/bignumber from 5.2.0 to 5.4.0

Bumps [@ethersproject/bignumber](https://github.com/ethers-io/ethers.js/tree/HEAD/packages/bignumber) from 5.2.0 to 5.4.0.
- [Release notes](https://github.com/ethers-io/ethers.js/releases)
- [Changelog](https://github.com/ethers-io/ethers.js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ethers-io/ethers.js/commits/v5.4.0/packages/bignumber)

---
updated-dependencies:
- dependency-name: "@ethersproject/bignumber"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump gulp-less from 4.0.1 to 5.0.0

Bumps [gulp-less](https://github.com/gulp-community/gulp-less) from 4.0.1 to 5.0.0.
- [Release notes](https://github.com/gulp-community/gulp-less/releases)
- [Commits](https://github.com/gulp-community/gulp-less/commits/5.0.0)

---
updated-dependencies:
- dependency-name: gulp-less
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @ethersproject/units from 5.2.0 to 5.4.0

Bumps [@ethersproject/units](https://github.com/ethers-io/ethers.js/tree/HEAD/packages/units) from 5.2.0 to 5.4.0.
- [Release notes](https://github.com/ethers-io/ethers.js/releases)
- [Changelog](https://github.com/ethers-io/ethers.js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ethers-io/ethers.js/commits/v5.4.0/packages/units)

---
updated-dependencies:
- dependency-name: "@ethersproject/units"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
updated apollo client to use graph ql in apolloClient.js

Commit:
added darktheme.js

Commit:
added darktheme.js

Commit:
started testing paper and typography on dashboard card

Commit:
started testing paper and typography on dashboard card

drondin committed:
Fix

drondin committed:
fix var name

drondin committed:
Add fetch

drondin committed:
working!

drondin committed:
update function

drondin committed:
remove parameters

Commit:
dashboard components updated, sidebar started, main layout in place

unbanksy committed:
Add unstkae allowance during account load.

unbanksy committed:
Remove deprecated method.o

Commit:
sidebar almost finished, just need to overide link colors and hover styles, stake page started

Commit:
updated stake page to use paper and Button components, still need to override hover styles

dependabot[bot] committed:
Bump ipfs-http-client from 45.0.0 to 50.1.2

Bumps [ipfs-http-client](https://github.com/ipfs/js-ipfs) from 45.0.0 to 50.1.2.
- [Release notes](https://github.com/ipfs/js-ipfs/releases)
- [Changelog](https://github.com/ipfs/js-ipfs/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ipfs/js-ipfs/compare/ipfs-http-client@45.0.0...ipfs-http-client@50.1.2)

---
updated-dependencies:
- dependency-name: ipfs-http-client
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump ethers from 5.2.0 to 5.4.0

Bumps [ethers](https://github.com/ethers-io/ethers.js/tree/HEAD/packages/ethers) from 5.2.0 to 5.4.0.
- [Release notes](https://github.com/ethers-io/ethers.js/releases)
- [Changelog](https://github.com/ethers-io/ethers.js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ethers-io/ethers.js/commits/v5.4.0/packages/ethers)

---
updated-dependencies:
- dependency-name: ethers
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
bond page components, stake page components, button and paper implemented still need to change typography and links

dependabot[bot] committed:
Bump dotenv from 8.6.0 to 10.0.0

Bumps [dotenv](https://github.com/motdotla/dotenv) from 8.6.0 to 10.0.0.
- [Release notes](https://github.com/motdotla/dotenv/releases)
- [Changelog](https://github.com/motdotla/dotenv/blob/master/CHANGELOG.md)
- [Commits](https://github.com/motdotla/dotenv/compare/v8.6.0...v10.0.0)

---
updated-dependencies:
- dependency-name: dotenv
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump react-scripts from 3.4.4 to 4.0.3

Bumps [react-scripts](https://github.com/facebook/create-react-app/tree/HEAD/packages/react-scripts) from 3.4.4 to 4.0.3.
- [Release notes](https://github.com/facebook/create-react-app/releases)
- [Changelog](https://github.com/facebook/create-react-app/blob/master/CHANGELOG-3.x.md)
- [Commits](https://github.com/facebook/create-react-app/commits/react-scripts@4.0.3/packages/react-scripts)

---
updated-dependencies:
- dependency-name: react-scripts
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
sidebar spacing, mobile bond views, typography. anext up link colors and table format

Commit:
sidebar responsiveness tweaks and link styling, stake table replaced with mui table

Commit:
small tweaks

Commit:
dashboard top card header cards adjusted and mobile tile view centered

Commit:
fixed links in bond discounts

Commit:
top bar buttons basic theme applied

ohmie33 committed:
Added new sOHM/Ohm mm logos and fixed layout bug in big screens

ohmie33 committed:
Use addresses from constant and use FRAX for uniswap

ohmie33 committed:
remove logs

ohmie33 committed:
Added token image helper and fixed eslint issues

Commit:
links and styles updated for bond table

dependabot[bot] committed:
Bump postcss from 8.3.0 to 8.3.5

Bumps [postcss](https://github.com/postcss/postcss) from 8.3.0 to 8.3.5.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.3.0...8.3.5)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @testing-library/user-event from 12.8.3 to 13.1.9

Bumps [@testing-library/user-event](https://github.com/testing-library/user-event) from 12.8.3 to 13.1.9.
- [Release notes](https://github.com/testing-library/user-event/releases)
- [Changelog](https://github.com/testing-library/user-event/blob/master/CHANGELOG.md)
- [Commits](https://github.com/testing-library/user-event/compare/v12.8.3...v13.1.9)

---
updated-dependencies:
- dependency-name: "@testing-library/user-event"
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eslint from 6.8.0 to 7.29.0

Bumps [eslint](https://github.com/eslint/eslint) from 6.8.0 to 7.29.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v6.8.0...v7.29.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

ohmie33 committed:
Use networkID

Commit:
styled top bar/buttons, copied over light and girth theme (need work) and removed more redundnacy

Commit:
removed theme file and broke into themes folder. deprecated styled components

Commit:
updated primary outlined button styles

Commit:
contained secondary button style tweak

Commit:
theme toggle styled, bonds page basic styles, fixed rounded sidebar issue

Commit:
stake

Commit:
updated eslint to 7.29 and changed prettier config to warn

drondin committed:
Connect subgraph by name

Commit:
commented out airbnb in eslint

Commit:
autoformat test

Commit:
fixed dep issues, updated formatting, styled mobile nav, styled migrate page

ohmie33 committed:
Added more staking info and UI fixes

ohmie33 committed:
Make sidebar discount numbers bold

dependabot[bot] committed:
Bump @testing-library/react from 11.2.7 to 12.0.0

Bumps [@testing-library/react](https://github.com/testing-library/react-testing-library) from 11.2.7 to 12.0.0.
- [Release notes](https://github.com/testing-library/react-testing-library/releases)
- [Changelog](https://github.com/testing-library/react-testing-library/blob/main/CHANGELOG.md)
- [Commits](https://github.com/testing-library/react-testing-library/compare/v11.2.7...v12.0.0)

---
updated-dependencies:
- dependency-name: "@testing-library/react"
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
top bar nearly done, sidebar refactored (mostly) to use material ui drawer, bootstrap removed, sidebar styled, typography implemented

Commit:
fixed topbar, stake mobile buttons, bond view, bond modal

Commit:
formatting

georgeroman committed:
Alert user if connected to the wrong network

ohmygirth committed:
imported new icons (still need to implement), cformatted files to clear prettier warnings, still need to fix advanced settings and style input fields

ohmygirth committed:
removed some unneeded pngs and updated social.jsx to use new components and materal svgicon component

ohmie33 committed:
153: adds current index following new design

ohmygirth committed:
link hover styles no underline

ohmygirth committed:
removed unneeded rebase timer styles, replaced bond cancel with styled icon, began styling for bond settings. main things left are new icons and input fields

ohmygirth committed:
imported new icons and got them working with theme colors

ohmygirth committed:
updated spacing on snav content and svgs

ohmygirth committed:
replaced input on stake page and styled some

acidpunk0x committed:
Update bug_report.md (#228)

Added section for possible workarounds

Zayen-X committed:
updated to ohmCirculatingSupply

Zayen-X committed:
Linting fixes

Zayen-X committed:
npm run lint passes

Zayen-X committed:
change alert to log

Zayen-X committed:
short circuit provider not existing. feels bad

ohmygirth committed:
refactored bond views

ohmygirth committed:
improved stake page styling

ohmygirth committed:
cleaned up topbar, made hamburger left anchored, removed font awesome for custom icons

ohmygirth committed:
top bar ohm info button integrated and refactored

ohmygirth committed:
staking updated with current index, button links fixed, bond cards styled

ohmygirth committed:
staking view cleaned up, still needs some refactoring but fine for now

ohmygirth committed:
added reward amount

ohmygirth committed:
good stopping point

ohmygirth committed:
rebased from develop. everything appears to work except rebase timer

ohmygirth committed:
fixed scroll

ohmygirth committed:
styled active buttons for dark theme, fixed sidebar spacing, updated stake pohm frax, improved top bar spacind for mobile

ohmygirth committed:
farm pools -> farm pool

ohmygirth committed:
top bar positioning and disapearing ohm button fixed

ohmygirth committed:
last tweaks to hades and card styles

ohmygirth committed:
fixed overflow on dashboard

ohmygirth committed:
girth mode fixes

ohmygirth committed:
aminor color tweaks

ohmygirth committed:
added fix for mobile topbar button styles, updated light themes to be closer to spec

ohmygirth committed:
fixed  next rebase issue

ohmygirth committed:
added basic card animations

ohmygirth committed:
resolving conflicts

ohmygirth committed:
fixed merge breaks

ohmygirth committed:
nof course fleek doesnt want to biuild it now

ohmygirth committed:
fixing stake styles that got messed up

ohmygirth committed:
fixing dashboard styles and installing node-sass

ohmygirth committed:
attempted hook optimization on theme toggle and slight refactoring

ohmygirth committed:
removed unused imports

dwjanus committed:
Merge pull request #234 from OlympusDAO/v1_2-169-cleanup-one

V1 2 169 cleanup one

unbanksy commented on src/actions/App.actions.js:
```diff
@@ -113,6 +113,8 @@ export const loadAppDetails =
     // Current index
     const currentIndex = await stakingContract.index();
 
+    console.log("graphData", graphData);
```
Delet ser

dwjanus replied:
kek howd that get n ther. deleting.

unbanksy commented on src/components/BondLogo.jsx:
```diff
@@ -1,21 +1,20 @@
 function BondHeader({ bond }) {
   const ohmAssetImg = () => {
-    return "https://raw.githubusercontent.com/sushiswap/assets/master/blockchains/ethereum/assets/0x383518188C0C6d7730D91b2c03a03C837814a899/logo.png";
+    return getOhmTokenImage();
```
Probably generalizes to `getTokenImage(argument)` where argument is a string

dwjanus replied:
yea i think i suggested this on another pr lol. will improve the helper in next update.

unbanksy commented on src/components/TopBar/OhmMenu.jsx:
```diff
@@ -0,0 +1,106 @@
+import { ReactComponent as ArrowUpIcon } from "../../assets/icons/v1.2/arrow-up.svg";
+import "./ohmmenu.scss";
+
+const TOKEN_DECIMALS = 9;
```
I personally prefer keeping global variables in a single file (e.g. constants) and then importing.

dwjanus replied:
facts. this is likely from copypasta of the original code. making a note to include this in next update

unbanksy commented on src/themes/girth.js:
```diff
@@ -0,0 +1,174 @@
+  containedSecondaryButtonHoverBG: "#363840",
+};
+
+export const girth = responsiveFontSizes(
```
kek

unbanksy commented on src/views/Bond/BondRedeem.jsx:
```diff
@@ -0,0 +1,111 @@
+import { useSelector, useDispatch } from "react-redux";
```
@dwjanus when I visit the Frax bond page, the Redeem V1 is not clickable. I don't think we want to show it, right?

![image](https://user-images.githubusercontent.com/81924824/125698198-2e5e3d3f-511b-46b3-b4cc-671888559e1c.png)

dwjanus replied:
i think youre right. I wasnt positive on if we were removing it so I set it to disabled while I got clarification.

unbanksy reviewed (approved):
@dwjanus this is 🔥 . Good call on material-ui. Let's :shipit:

dwjanus commented:
Major Changes 

1. removed bootstrap and rebuilt Sidebar with Material UI
2. deleted a majority of unused files from scaffold (kept hooks and a couple components that looked potentially useful)
3. removed styled components for original theming and replaced with material ui custom themes (see /themes)
4. Refactored Bonds view
5. Replaced majority of html in jsx files with Material UI counterparts
6. Implemented Typography (Material UI)
7. Refactored TopBar using Material UI and added OhmMenu for dropdown
8. Removed/refactored majority of redundant or conflicting scss 
9. Implemented major design spec requirements within Material UI theme definitions 
10. removed font-awesome and replaced icons with custom icons from figma

ohmygirth committed:
refactored images in helper, moved constants out of ohmmenu, removed console logs, fixed bond/bond settings background colors

dwjanus committed:
Merge pull request #235 from OlympusDAO/v1_2-169-cleanup-one

refactored images in helper, moved constants out of ohmmenu, removed …
